src/box/lua/config/init.lua

```diff
@@ -95,15 +95,20 @@ local function broadcast(self)
     box.broadcast('config.info', self:info())
 end
 
-function methods._alert(self, alert)
+function methods._alert(self, key, alert)
     assert(alert.type == 'error' or alert.type == 'warn')
     if alert.type == 'error' then
         log.error(alert.message)
     else
         log.warn(alert.message)
     end
     alert.timestamp = datetime.now()
-    table.insert(self._alerts, alert)
+    self._alerts[key] = alert
+end
+
+function methods._alert_drop(self, key)
+    self._alerts[key] = nil
+    self:_set_status_based_on_alerts()
```

Comment:
I'm not sure that we should set the status right after any alert drop. What if we're in `reload_in_progress` status? It'll change it to `ready`, `check_warnings` or `check_errors` before the reload will actually end.

The `_set_status_based_on_alerts()` should be called in certain points, because the status is to be changed after certain events or actions.

Reply:
OTOH, `check_warnings` -> `ready` transition is completely OK if the last warning was removed and there are no errors.

However, I wouldn't go from `check_errors` and wouldn't drop errors without a reload, because a first error during a reload stops the reloading process. Some parts of the new configuration may be not applied.

Reply:
Fixed